.aws/faucet.sh

```diff
@@ -19,14 +19,25 @@ do
             DESTINATION=$(echo $row | awk '{print $4}')
             if [[ ${#DESTINATION} -eq 45 ]];
             then
-                # work only on stuff that have the right length
-                ADDRESS=$(./gaiacli keys show ${ACCOUNT} --home . --address)
-                echo ${PASSWORD} | ./gaiacli tx send ${DESTINATION} ${AMOUNTS} --home . --from ${ADDRESS} --chain-id=${NETWORK}
-                echo ${PASSWORD} | ./gaiacli tx send ${DESTINATION} ${AMOUNTP} --home . --from ${ADDRESS} --chain-id=${NETWORK}
-                echo ${PASSWORD} | ./gaiacli tx send ${DESTINATION} ${AMOUNTC} --home . --from ${ADDRESS} --chain-id=${NETWORK}
+                ACCOUNT_INFO=$(./gaiacli query account ${DESTINATION} --chain-id ${NETWORK} --trust-node --home .)
+                dt=$(date '+%d/%m/%Y %H:%M:%S');
+                if [[ ${ACCOUNT_INFO} == *"stake"* ]]; then
+                    echo "$dt - $DESTINATION already funded, bye greedy b****rd!"
+                else
+                    # work only on stuff that have the right length
+                    ADDRESS=$(./gaiacli keys show ${ACCOUNT} --home . --address)
+                    echo "stakes $AMOUNTS at $DESTINATION"
+                    echo ${PASSWORD} | ./gaiacli tx send ${DESTINATION} ${AMOUNTS} --home . --from ${ADDRESS} --chain-id=${NETWORK}
+                    sleep 5s # TODO: should be smarter, check if block was created or not
```

Comment:
smaller timeout doesn't work? @alessio this was the problem I explained

Reply:
yeah in this moment apparently that's the safest solution available